Makefile

```diff
@@ -12,23 +12,36 @@ bin/boot.o: src/boot/boot.asm | bin
 		-o $@ \
 		$<
 
-# -O ReleaseFast
-bin/boot_zig.o: src/boot/boot.zig | bin
-	zig build-obj -target x86_64-freestanding-gnu -static -I./src/kernel -mno-red-zone -femit-bin=$@ $<
+PKGS=--pkg-begin zasm ./src/zasm.zig --pkg-end
+
+bin/%_32_zig.o: src/boot/%_32.zig | bin
+	@zig build-obj \
+		--cache-dir bin/zig-cache \
+		-I src/kernel \
+		-mno-red-zone \
+		-static \
+		-target i386-freestanding-gnu \
+		${PKGS} \
+		-O ReleaseFast \
+		-femit-bin=$@ \
+		$<
+
+bin/%_elf64_zig.o: bin/%_32_zig.o
+	@objcopy --output-target elf64-x86-64 $< $@
```

Comment:
Hack-hackli-hack

Reply:
Maybe add a todo to remove this when we can solve this in zig?

Reply:
Sure! You don't like hack-hackiilihack?? ;P

Reply:
Done dc2d8cbc410e99b1cb35372e712e87f0c1c612b7.